src/phpDocumentor/ConfigurationFactory/PhpDocumentor3.php

```diff
@@ -0,0 +1,197 @@
+<?php
+/**
+ * This file is part of phpDocumentor.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @copyright 2010-2015 Mike van Riel<mike@phpdoc.org>
+ * @license   http://www.opensource.org/licenses/mit-license.php MIT
+ * @link      http://phpdoc.org
+ */
+
+namespace phpDocumentor\ConfigurationFactory;
+
+/**
+ * phpDocumentor3 strategy for converting the configuration xml to an array.
+ */
+final class PhpDocumentor3 implements Strategy
+{
+    /**
+     * The path to the xsd that is used for validation of the configuration file.
+     *
+     * @var string
+     */
+    private $schemaPath;
+
+    /**
+     * Initializes the PhpDocumentor3 strategy.
+     *
+     * @param string $schemaPath
+     */
+    public function __construct($schemaPath)
+    {
+        if ($schemaPath === '') {
+            $schemaPath = __DIR__ . '../../../../data/xsd/phpdoc.xsd';
+        }
+
+        $this->schemaPath = $schemaPath;
+    }
+
+    /**å
+     * Converts the phpDocumentor3 configuration xml to an array.
+     *
+     * @param \SimpleXMLElement $phpDocumentor
+     *
+     * @return array
+     */
+    public function convert(\SimpleXMLElement $phpDocumentor)
+    {
+        $this->validate($phpDocumentor);
+
+        $versions = [];
+        $template = [];
+
+        foreach ($phpDocumentor->children() as $key => $value) {
+            switch ((string) $key) {
+                case 'version':
+                    $versions[(string) $value->attributes()->number] = $this->buildVersions($value);
+                    break;
+                case 'template':
+                    $template = $this->buildTemplate($value);
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        $outputDirectory = ((string) $phpDocumentor->parser->target) ?: 'file://build/docs';
+        $cacheDirectory  = ((string) $phpDocumentor->parser->cache) ?: '/tmp/phpdoc-doc-cache';
+
+        $phpdoc3Array = [
+            'phpdocumentor' => [
+                'paths'     => [
+                    'output' => $outputDirectory,
+                    'cache'  => $cacheDirectory,
+                ],
+                'versions'  => $versions,
+                'templates' => $template,
+            ],
+        ];
+
+        return $phpdoc3Array;
+    }
+
+    /**
+     * Checks if the provided configuration xml is for phpDocumentor3.
+     *
+     * @param $phpDocumentor
+     *
+     * @return bool
+     */
+    public function match(\SimpleXMLElement $phpDocumentor)
+    {
+        return (int) $phpDocumentor->attributes()->version === 3;
```

Comment:
Please do not convert it into an int but do a string comparison; if we want to introduce another parser for 3.1 then the adapter for 3.1 is not selected but this on instead (`(int)"3.1" === 3`)